### PR TITLE
hnerv_muon submission (0.20)

### PR DESCRIPTION
**Pull request by AaronLeslie138**
# submission name:
hnerv_muon

# upload zipped `archive.zip`
[archive.zip](https://github.com/user-attachments/files/27332334/archive.zip)

# report.txt
```
=== Evaluation results over 600 samples ===
  Average PoseNet Distortion: 0.00003494
  Average SegNet Distortion: 0.00061212
  Submission file size: 178,417 bytes
  Original uncompressed size: 37,545,489 bytes
  Compression Rate: 0.00475202
  Final score: 100*segnet_dist + √(10*posenet_dist) + 25*rate = 0.20
```

# does your submission require gpu for evaluation (inflation)?
no (CPU works; faster on GPU)

# did you include the compression script? and want it to be merged?
yes. `compress.sh` and `src/` are included and the pipeline is reproducible from random init

# additional comments
229K-parameter HNeRV decoder + 28-d-per-frame-pair latents (~600 pairs), INT8-quantized and brotli-compressed to 178 KB total. The decoder is trained via an 8-stage curriculum: cross-entropy seg → τ-Softplus margin → smooth disagreement → +QAT → +L7 hard-pixel weighting + C1a regularizer → λ-sweep → σ-sweep → +Muon optimizer (with WD=5e-4 per Chen-Li-Liu's spectral-norm KKT theory). The C1a regularizer shapes the weight distribution toward the integer grid, which collapses the entropy floor for downstream brotli compression. Verified end-to-end via this repo's eval workflow on `ubuntu-latest` CPU; final score 0.1987

Full writeup: https://aaronleslie.dev/blog/comma-compression

### CODE REVIEW

**Comment by AaronLeslie138** [comment]
Wow! Woke up to no shortage of people fine-tuning this in the last 3 hours to eek out an improvement on top. Might have to re-title my blog post  ; ) Regrettable that the landscape punishes publishing /src, but cool to see the small improvements folks were able make. Heart on my sleeve, it does suck to lose exclusively to submissions that - code is one thing - just used my archive.zip contents and not a fresh base model. The competition seems to mainly reward whoever is up using codex or claude right at the last hour, not the author of 99% of the work. `</vent>`

**Comment by YassineYousfi** [comment]
`<re:vent>` we are going to reward folks publishing their code even if not in top 3

**Comment by AaronLeslie138** [comment]
Is `0.23` a cuda discrepancy? I ran eval.yml on my own repo on `ubuntu-latest` without a GPU (since that was free) to get the report.txt in the body of this PR. I would expect it to be 1:1 identical, since I just used a cloned Github Action. (Unless I've made an error somewhere, but I see similar cases for other folks who used the same approximate hnerv solution where the `eval` is higher on cuda).

**Comment by YassineYousfi** [comment]
yeah there is a small hw difference in the decode, I ran all submissions in t4 for a fair comparison.

**Comment by YassineYousfi** [comment]
great submission and write-up!

**Comment by EthanYangTW** [comment]
@AaronLeslie138 from what I see from all the submisssion,you definitely won
I think you definitely deserved a price Awesome work mate

**Comment by YassineYousfi** [comment]
@AaronLeslie138 This submission won an honorable prize and best write-up prize. Please email me at {first name}@comma.ai for logistics.
Let us know if you are looking for a job/internship as well.
Congratulations!

**Comment by YassineYousfi** [comment]
small typo in your write-up title "Spritually" -> "Spiritually"

**Comment by AaronLeslie138** [comment]
> small typo in your write-up title "Spritually" -> "Spiritually"

Thanks for catching this! I slipped that in to the title quickly on Monday after I woke up and didn't proofread myself. And thanks again for hosting this competition! It was an interesting problem
